grpc.go

```diff
@@ -1449,3 +1449,26 @@ func (a *agentGRPC) SetGuestDateTime(ctx context.Context, req *pb.SetGuestDateTi
 	}
 	return &gpb.Empty{}, nil
 }
+
+func (a *agentGRPC) CopyFile(ctx context.Context, req *pb.CopyFileRequest) (*gpb.Empty, error) {
+	if err := os.MkdirAll(filepath.Dir(req.Path), os.FileMode(req.DirMode)); err != nil {
+		return emptyResp, err
+	}
+
+	file, err := os.OpenFile(req.Path, os.O_WRONLY|os.O_CREATE, os.FileMode(req.FileMode))
```

Comment:
- What happens if the file exists as, say, a socket? Won't this overwrite it as a regular file? Maybe we should check to ensure the types are the same?

- Maybe we should enhance the `getAgentDetails()`gRPC API to include the architecture "just in case" to handle the scenario where the caller wants to send a binary file?

- I don't think we should just blindly create a file with the mode specified. What if this is a long-running container and the `CopyFile()` ends up opening ports somehow or overwriting critical files, or making critical file world-writable...